posthog/hogql_queries/web_analytics/test/test_web_overview.py

```diff
@@ -185,3 +197,10 @@ def test_correctly_counts_pageviews_in_long_running_session(self, use_sessions_t
 
         sessions = results[2]
         self.assertEqual(1, sessions.value)
+
+    @patch("posthog.hogql.query.sync_execute", wraps=sync_execute)
+    def test_limit_is_context_aware(self, mock_sync_execute: MagicMock):
+        self._run_web_overview_query("2023-12-01", "2023-12-03", limit_context=LimitContext.QUERY_ASYNC)
+
+        mock_sync_execute.assert_called_once()
+        self.assertIn(f" max_execution_time={INCREASED_MAX_EXECUTION_TIME},", mock_sync_execute.call_args[0][0])
```

Comment:
It's really tedious and brittle that we don't have a way to add a test for common mechanics to ALL query runner suites. `limit_context`-awareness is one thing that needs to be guaranteed for _each_ implementation, as they implement their own `calculate()`s.

It looks like we could use a `QueryRunnerTest` base class that would inherit from `ClickhouseTestMixin` and have `run_query()` and `run_actors_query()`. Was there a reason for not doing that? @Gilbert09 @webjunkie

Reply:
I guess it was because of the conversion of tests basically is copying one by one the legacy tests and nobody worried about combining code or functionality yet 🙈

Reply:
👍 Cool, makes sense. I don't have the time to do this right now, but useful to do in some nearish future